memcached.c

```diff
@@ -0,0 +1,6177 @@
+        char *temp = strdup(slab_sizes_unparsed);
+        if (_parse_slab_sizes(slab_sizes_unparsed, slab_sizes)) {
+            use_slab_sizes = true;
+            if (meta->slab_config) {
+                free(meta->slab_config);
+            }
+            meta->slab_config = temp;
+        } else {
+            exit(EX_USAGE);
+        }
+    } else if (!meta->slab_config) {
+        // using the default factor.
+        meta->slab_config = "1.25";
+    }
+
+    if (settings.hot_lru_pct + settings.warm_lru_pct > 80) {
+        fprintf(stderr, "hot_lru_pct + warm_lru_pct cannot be more than 80%% combined\n");
+        exit(EX_USAGE);
+    }
+
+    if (settings.temp_lru && !start_lru_maintainer) {
+        fprintf(stderr, "temporary_ttl requires lru_maintainer to be enabled\n");
+        exit(EX_USAGE);
+    }
+
+    if (hash_init(hash_type) != 0) {
+        fprintf(stderr, "Failed to initialize hash_algorithm!\n");
+        exit(EX_USAGE);
+    }
+
+    /*
+     * Use one workerthread to serve each UDP port if the user specified
+     * multiple ports
+     */
+    if (settings.inter != NULL && strchr(settings.inter, ',')) {
+        settings.num_threads_per_udp = 1;
+    } else {
+        settings.num_threads_per_udp = settings.num_threads;
+    }
+
+    if (settings.sasl) {
+        if (!protocol_specified) {
+            settings.binding_protocol = binary_prot;
+        } else {
+            if (settings.binding_protocol != binary_prot) {
+                fprintf(stderr, "ERROR: You cannot allow the ASCII protocol while using SASL.\n");
+                exit(EX_USAGE);
+            }
+        }
+
+        if (settings.udpport) {
+            fprintf(stderr, "ERROR: Cannot enable UDP while using binary SASL authentication.\n");
+            exit(EX_USAGE);
+        }
+    }
+
+    if (settings.auth_file) {
+        if (!protocol_specified) {
+            settings.binding_protocol = ascii_prot;
+        } else {
+            if (settings.binding_protocol != ascii_prot) {
+                fprintf(stderr, "ERROR: You cannot allow the BINARY protocol while using ascii authentication tokens.\n");
+                exit(EX_USAGE);
+            }
+        }
+    }
+
+    if (udp_specified && settings.udpport != 0 && !tcp_specified) {
+        settings.port = settings.udpport;
+    }
+
+
+#ifdef TLS
+    /*
+     * Setup SSL if enabled
+     */
+    if (settings.ssl_enabled) {
+        if (!settings.port) {
+            fprintf(stderr, "ERROR: You cannot enable SSL without a TCP port.\n");
+            exit(EX_USAGE);
+        }
+        // openssl init methods.
+        SSL_load_error_strings();
+        SSLeay_add_ssl_algorithms();
+        // Initiate the SSL context.
+        ssl_init();
+    }
+#endif
+
+    if (maxcore != 0) {
+        struct rlimit rlim_new;
+        /*
+         * First try raising to infinity; if that fails, try bringing
+         * the soft limit to the hard.
+         */
+        if (getrlimit(RLIMIT_CORE, &rlim) == 0) {
```

Comment:
whats being done here? [@prashanthraghu](https://twitter.com/prashanthraghu)

We will use this for our discussion here, only thing is it migth take too much time to load this page.